soc/nxp/mcx/mcxnx4x/CMakeLists.txt

```diff
@@ -8,10 +8,11 @@
 # Flexcomm SDK driver
 zephyr_compile_definitions_ifdef(CONFIG_NXP_LP_FLEXCOMM LPFLEXCOMM_INIT_NOT_USED_IN_DRIVER=1)
 
-zephyr_sources(
-  soc.c
-  )
+zephyr_sources(soc.c)
+zephyr_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_XIP flash_clock_setup.c)
```

Comment:
This source still needs to be included if we are not using `CONFIG_FLASH_MCUX_FLEXSPI_XIP`, right? Since we still need to reclock the FlexSPI even if we are not running in XIP mode

Reply:
Thanks for catching this. Tested with the `flash_shell` to ensure we can read, write and erase.

Reply:
Thanks Daniel for the suggestion, I have update the Cmake file.